js/binance.js

```diff
@@ -5134,6 +5134,13 @@ module.exports = class binance extends Exchange {
             if ((error === '-2015') && this.options['hasAlreadyAuthenticatedSuccessfully']) {
                 throw new DDoSProtection (this.id + ' temporary banned: ' + body);
             }
+            if (message === 'No need to change margin type.') {
+                // not an error
+                // https://github.com/ccxt/ccxt/issues/11268
+                // https://github.com/ccxt/ccxt/pull/11624
+                // POST https://fapi.binance.com/fapi/v1/marginType 400 Bad Request             // binanceusdm {"code":-4046,"msg":"No need to change margin type."}
+                return undefined;
```

Comment:
Returning `undefined` will actually ignore that message. Apparently, their exception codes differ between sapi and fapi. So we will have to subdivide the exception mappings further by the type of the api.
The `NoChange` name is too short for this specific exception, I would suggest to avoid it for now.
Let's just return `undefined` from the error handler when it's a fapi-endpoint and the error code is '-4046' (that would be faster than comparing the message).

Reply:
It looks like `-4046` is an error code for something else though

![](https://i.imgur.com/uoq6lFp.png)

Reply:
I wasn't using that new error code I created, so returning undefined does nothing in this case

Reply:
Yes, their codes differ between spot apis and swap apis, so we will probably add `exchange.exceptions['exact'][defaultType] = { ... }` to differentiate between them. In the meantime I suggest to detect it like this: 
1. in `handleErrors` check if it's `fapi` and if the code is `-4046` and return `undefined` if it is
2. we don't have to intercept it in the fetch-method, the user will simply get the response containing the exchange message with code -4046, as is.

Reply:
There's no way to check if it's fapi without breaking apart the url, what's wrong with checking the message? It works, and undefined is returned, but an error gets thrown anyway

![](https://i.imgur.com/BoMG6Ej.gif)

Reply:
> It works, and undefined is returned, but an error gets thrown anyway

Yep, if handleErrors does not throw, the default error handler is called as well. And that checks the http response code in the httpExceptions map. In this case the exchange also sends HTTP code 400, which triggers the lower lever default error handler, and the error is still thrown. Apparently, we need a way of indicating whether further error handling should be performed or skipped. What comes to my mind as the least intrusive solution to it – we can return `true` from `handleErrors` and that would mean `skip further error handling`. That would be easy to intercept.